src/django/api/models.py

```diff
@@ -2959,6 +2967,30 @@ class Meta():
     )
     objects = TrainedModelManager()
 
+    def activate(self):
+        if self.pk is None:
+            raise NoTrainedModel()
+        if self.is_active:
+            raise ModelAlreadyActive()
+        if self.activated_at is not None:
+            raise ModelAlreadyActive()
+        with transaction.atomic():
+            prev_active_version_id = TrainedModel.objects.get_active_version_id()
+            TrainedModel.objects.update(is_active=False)
+            self.is_active = True
+            self.activated_at = timezone.now()
+            self.save()
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    """
+                        ALTER TABLE dedupe_indexed_records RENAME TO dedupe_indexed_records_{prev_active_version_id};
+                        ALTER TABLE dedupe_indexed_records_{active_version_id} RENAME TO dedupe_indexed_records;
+                        """.format({
```

Comment:
I was getting a `KeyError` exception on this line. I did a search and ended up here https://stackoverflow.com/a/5952472. When I added the `**` operator it correctly unpacked the dictionary into keyword arguments.

```suggestion
                        """.format(**{
```